playback/src/player.rs

```diff
@@ -1191,6 +1206,108 @@ impl PlayerTrackLoader {
             });
         }
     }
+
+    async fn load_local_track(
+        &self,
+        track_uri: SpotifyUri,
+        position_ms: u32,
+    ) -> Option<PlayerLoadedTrackData> {
+        let entry = self.local_file_lookup.get(&track_uri);
+
+        let Some(path) = entry else {
+            error!("Unable to find file path for local file <{track_uri}>");
+            return None;
+        };
+
+        debug!(
+            "Located file path '{}' for local file <{track_uri}>",
+            path.display()
+        );
+
+        let src = match File::open(path) {
+            Ok(src) => src,
+            Err(e) => {
+                error!("Failed to open local file: {e}");
+                return None;
+            }
+        };
+
+        let decoder = match SymphoniaDecoder::new_with_probe(
+            src,
+            path.extension().and_then(|e| e.to_str()),
+        ) {
+            Ok(decoder) => decoder,
+            Err(e) => {
+                error!("Error decoding local file: {e}");
+                return None;
+            }
+        };
+
+        let mut decoder = Box::new(decoder);
+        let normalisation_data = decoder.normalisation_data().unwrap_or_else(|| {
+            warn!("Unable to get normalisation data, continuing with defaults.");
+            NormalisationData::default()
+        });
+
+        let local_file_metadata = decoder.local_file_metadata().unwrap_or_default();
+
+        let stream_position_ms = match decoder.seek(position_ms) {
+            Ok(new_position_ms) => new_position_ms,
+            Err(e) => {
+                error!(
+                    "PlayerTrackLoader::load_local_track error seeking to starting position {position_ms}: {e}"
+                );
+                return None;
+            }
+        };
+
+        let SpotifyUri::Local { duration, .. } = track_uri else {
+            error!("Unable to determine track duration for local file: not a local file URI");
+            return None;
+        };
```

Comment:
It probably makes more sense to move this to the top of the method so we don't do everything beforehand, just to discard it all here^^;

Reply:
Yep, you're right, moved